README.md

```diff
@@ -1,3 +1,236 @@
+│   └── cli.py              # Command-line interface
+├── tests/                   # Test suite
+├── generate_website.py      # Main site generator
+└── Makefile                # Build commands
+```
+
+## Contributing FAQ Entries
+
+We welcome FAQ contributions! There are two ways to contribute:
+
+### 1. Automated (Recommended)
+
+Simply [create a new issue](../../issues/new/choose) using the "FAQ Proposal" template. Our FAQ Bot will:
+
+1. Analyze your proposal against existing FAQs
+2. Decide the best action (NEW, UPDATE, or DUPLICATE)
+3. Either create a PR with your changes or provide feedback
+
+See [CONTRIBUTING.md](CONTRIBUTING.md) for detailed instructions.
+
+### 2. Manual Pull Request
+
+1. Fork this repository
+2. Add/modify FAQ files in `_questions/{course}/{section}/`
+3. Follow the frontmatter format (see below)
+4. Create a Pull Request
+
+## FAQ File Format
+
+Each FAQ entry is a markdown file with YAML frontmatter:
+
+```markdown
+---
+id: abc1234567
+question: 'How do I install the dependencies?'
+sort_order: 10
+---
+
+To install dependencies, run:
+
+\`\`\`bash
+uv pip install -r requirements.txt
+\`\`\`
+
+See the [installation guide](link) for more details.
+```
+
+### Frontmatter Fields
+
+- **id**: Unique 10-character identifier (auto-generated)
+- **question**: The FAQ question text
+- **sort_order**: Integer controlling order within section
+- **images** (optional): Array of image metadata for embedded images
+
+## FAQ Automation
+
+The FAQ automation system uses:
+
+- **minsearch**: Lightweight text search to find similar existing FAQs
+- **OpenAI GPT-4**: LLM-based decision making for triage
+- **Pydantic**: Structured output validation
+- **GitHub Actions**: Automated workflow orchestration
+
+### How It Works
+
+1. **Proposal Submission**: User creates GitHub issue with FAQ proposal
+2. **Retrieval**: System searches existing FAQs for similar content
+3. **LLM Analysis**: GPT-4 analyzes proposal + search results and decides:
+   - **NEW**: Question not covered → create new FAQ file
+   - **UPDATE**: Adds valuable context → merge with existing FAQ
+   - **DUPLICATE**: Already fully answered → close with explanation
+4. **Action Execution**:
+   - NEW/UPDATE: Create PR with file changes
+   - DUPLICATE: Comment on issue and close
+
+### Configuration
+
+The workflow requires:
+
+- **OPENAI_API_KEY**: Set as repository secret for GitHub Actions
+- **GitHub permissions**: `contents: write`, `issues: write`, `pull-requests: write`
+
+## Development
+
+### Setup
+
+```bash
+# Install dependencies
+uv pip install -e .
+
+# Install dev dependencies
+uv pip install -e ".[dev]"
+```
+
+### Running Locally
+
+```bash
+# Generate website
+python generate_website.py
+
+# Process a FAQ proposal (for testing)
+python -m faq_automation.cli \
+  --issue-body "$(cat test_issue.txt)" \
+  --issue-number 42 \
+  --course machine-learning-zoomcamp
+```
+
+### Testing
+
+```bash
+# Unit tests
+pytest tests/unit/ -v
```

Comment:
uv run ...

Reply:
Fixed by referencing commands in `Makefile` instead (e.g. `make test-unit`). I also included a comment referring the reader to the `tests/README.md` file for more details regarding testing.